sdk/cosmos/azure-cosmos/azure/cosmos/aio/container.py

```diff
@@ -0,0 +1,714 @@
+        )
+
+        if response_hook:
+            response_hook(self.client_connection.last_response_headers, self._properties)
+
+        return cast('Dict[str, Any]', self._properties)
+
+    @distributed_trace_async
+    async def create_item(
+        self,
+        body,  # type: Dict[str, Any]
+        pre_trigger_include=None,  # type: Optional[str]
+        post_trigger_include=None,  # type: Optional[str]
+        indexing_directive=None,  # type: Optional[Any]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> Dict[str, Any]
+        """Create an item in the container.
+
+        To update or replace an existing item, use the
+        :func:`ContainerProxy.upsert_item` method.
+
+        :param body: A dict-like object representing the item to create.
+        :param pre_trigger_include: trigger id to be used as pre operation trigger.
+        :param post_trigger_include: trigger id to be used as post operation trigger.
+        :param indexing_directive: Indicate whether the document should be omitted from indexing.
+        :keyword bool enable_automatic_id_generation: Enable automatic id generation if no id present.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword str etag: An ETag value, or the wildcard character (*). Used to check if the resource
+            has changed, and act according to the condition specified by the `match_condition` parameter.
+        :keyword ~azure.core.MatchConditions match_condition: The match condition to use upon the etag.
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :returns: A dict representing the new item.
+        :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: Item with the given ID already exists.
+        :rtype: dict[str, Any]
+        """
+        request_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
+
+        request_options["disableAutomaticIdGeneration"] = not kwargs.pop('enable_automatic_id_generation', False)
+        if pre_trigger_include is not None:
+            request_options["preTriggerInclude"] = pre_trigger_include
+        if post_trigger_include is not None:
+            request_options["postTriggerInclude"] = post_trigger_include
+        if indexing_directive is not None:
+            request_options["indexingDirective"] = indexing_directive
+
+        result = await self.client_connection.CreateItem(
+            database_or_container_link=self.container_link, document=body, options=request_options, **kwargs
+        )
+        if response_hook:
+            response_hook(self.client_connection.last_response_headers, result)
+        return result
+
+    @distributed_trace_async
+    async def read_item(
+        self,
+        item,  # type: Union[str, Dict[str, Any]]
```

Comment:
I see two options here:
1. we pass item as json and SDK infers both id and partitionKey
2. we pass itemId as string and partitionKey explicitly.

We should not mix the two options. same for the other APIs as well. We should look into this after the first preview.